src/main/java/gregtech/common/metatileentities/converter/ConverterTrait.java

```diff
@@ -0,0 +1,268 @@
+package gregtech.common.metatileentities.converter;
+
+import gregtech.api.GTValues;
+import gregtech.api.capability.FeCompat;
+import gregtech.api.capability.GregtechCapabilities;
+import gregtech.api.capability.IEnergyContainer;
+import gregtech.api.metatileentity.MTETrait;
+import gregtech.api.util.GTUtility;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.energy.CapabilityEnergy;
+import net.minecraftforge.energy.IEnergyStorage;
+
+public class ConverterTrait extends MTETrait {
+
+    private final int amps;
+    private final long voltage;
+
+    /**
+     * If TRUE, the front facing of the machine will OUTPUT EU, other sides INPUT FE.
+     *
+     * If FALSE, the front facing of the machine will OUTPUT FE, other sides INPUT EU.
+     */
+    private boolean feToEu;
+
+    private final IEnergyStorage energyFE = new FEContainer();
+    private final IEnergyContainer energyEU = new EUContainer();
+    protected long storedEU;
+
+    private final long baseCapacity;
+
+    private long usedAmps;
+
+    private BlockPos frontPos;
+
+    public ConverterTrait(MetaTileEntityConverter mte, int amps, boolean feToEu) {
+        super(mte);
+        this.amps = amps;
+        this.feToEu = feToEu;
+        this.voltage = GTValues.V[mte.getTier()];
+        this.baseCapacity = this.voltage * 16 * amps;
+    }
+
+    protected IEnergyContainer getEnergyEUContainer() {
+        return energyEU;
+    }
+
+    protected IEnergyStorage getEnergyFEContainer() {
+        return energyFE;
+    }
+
+    public boolean isFeToEu() {
+        return feToEu;
+    }
+
+    protected void setFeToEu(boolean feToEu) {
+        this.feToEu = feToEu;
+    }
+
+    public int getBaseAmps() {
+        return amps;
+    }
+
+    public long getVoltage() {
+        return voltage;
+    }
+
+    @Override
+    public String getName() {
+        return "EnergyConvertTrait";
+    }
+
+    @Override
+    public int getNetworkID() {
+        return 1;
```

Comment:
This NetworkID already belongs to a Trait. See MTETrait#TraitNetworkIds

Reply:
I think this is intentional to act as an energy container trait